Unable to build Windows arm64 export template 4.5-beta2 / SDL3

### Tested versions

- Godot 4.5-dev2 [`af2c71397`](https://github.com/godotengine/godot/commit/af2c713971499953373380b9ae8673f64423bd59) builds correctly
- Godot 4.5-dev3 [`28089c40c`](https://github.com/godotengine/godot/commit/28089c40c13597bf908802c61352c6fffe0a4465) fails with `error: use of undeclared identifier 'ACCESSKIT_HAS_POPUP_TRUE';`
- Godot 4.5-dev4 [`209a446e3`](https://github.com/godotengine/godot/commit/209a446e3657e6fd736b9b7589b94cbdaad2d854) fails with `error: use of undeclared identifier 'ACCESSKIT_HAS_POPUP_TRUE';`
- Godot 4.5-dev5 [`64b09905c`](https://github.com/godotengine/godot/commit/64b09905c7b2877f8aef99d8b63e73e5d31acfb9) fails with `error: use of undeclared identifier 'ACCESSKIT_HAS_POPUP_TRUE';`
- Godot 4.5-beta1 [`46c495ca2`](https://github.com/godotengine/godot/commit/46c495ca21f40f57a7fb9c7cde6143738f1652d4) builds correctly
- Godot 4.5-beta2 [`e1b4101e3`](https://github.com/godotengine/godot/commit/e1b4101e3460dd9c6ba0b7f8d88e9751b8383f5b) fails with error: `error: use of undeclared identifier 'EINVAL'`
- master [53be3b78d](https://github.com/godotengine/godot/commit/53be3b78d1634406f1fb29e3802c608a5f5104a1) fails with error: `error: use of undeclared identifier 'EINVAL'`


### System information

Ubuntu 24.04.2 LTS (x86_64)

### Issue description

I've built the container images using the [build-containers](https://github.com/godotengine/build-containers) and I'm compiling Godot with the [godot-build-scripts](https://github.com/godotengine/godot-build-scripts). 

Since I'm only interested in Windows builds, I've modified the `build.sh` and commented out all other platforms. The x86-64/32 builds compile fine, but once it gets to arm64, it will fail with the following error:

```
/root/llvm-mingw/bin/aarch64-w64-mingw32-clang -o thirdparty/sdl/joystick/hidapi/SDL_hidapi_rumble.windows.template_debug.arm64.llvm.o -c -std=gnu17 -ffp-contract=off -flto=thin -fcolor-diagnostics -O2 -w -DSDL_PLATFORM_PRIVATE -DDEBUG_ENABLED -DNDEBUG -DWINDOWS_ENABLED -DWASAPI_ENABLED -DWINMIDI_ENABLED -DWINVER=0x0A00 -D_WIN32_WINNT=0x0A00 -DACCESSKIT_ENABLED -DVULKAN_ENABLED -DRD_ENABLED -DSDL_ENABLED -DD3D12_ENABLED -DGLES3_ENABLED -DEGL_STATIC -DMINGW_ENABLED -DMINGW_HAS_SECURE_API=1 -DMINIZIP_ENABLED -DBROTLI_ENABLED -DTHREADS_ENABLED -DCLIPPER2_ENABLED -DZSTD_STATIC_LINKING_ONLY -DUSE_VOLK -DVK_USE_PLATFORM_WIN32_KHR -DDCOMP_ENABLED -DGLAD_ENABLED -DEGL_ENABLED -DSDL_PLATFORM_WINDOWS -Ithirdparty/sdl -Ithirdparty/sdl/include -Ithirdparty/sdl/include/build_config -Idrivers/sdl -I/root/accesskit/accesskit-c/include -Iplatform/windows -I. -isystem thirdparty/glad -isystem thirdparty/directx_headers/include/directx -isystem thirdparty/volk -isystem thirdparty/vulkan -isystem thirdparty/vulkan/include -isystem thirdparty/zstd -isystem thirdparty/zlib -isystem thirdparty/clipper2/include -isystem thirdparty/brotli/include -isystem thirdparty/angle/include -isystem thirdparty/d3d12ma thirdparty/sdl/joystick/hidapi/SDL_hidapi_rumble.c
thirdparty/sdl/stdlib/SDL_malloc.c:5313:14: error: use of undeclared identifier 'EINVAL'
 5313 |       return EINVAL;
      |              ^
thirdparty/sdl/stdlib/SDL_malloc.c:5321:12: error: use of undeclared identifier 'ENOMEM'
 5321 |     return ENOMEM;
      |            ^
2 errors generated.
scons: *** [thirdparty/sdl/stdlib/SDL_malloc.windows.template_debug.arm64.llvm.o] Error 1
scons: building terminated because of errors.
INFO: Time elapsed: 00:12:12.87
```

I can work around it by editing the `SDL_malloc.c` and ensuring that `#include <errno.h>` is included. The quickest 'hack' was to just move the line down by one outside the `LACKS_ERROR_H` check:

<img width="968" height="307" alt="Image" src="https://github.com/user-attachments/assets/f544a356-307c-4c07-9428-8acd344ceccb" />



### Steps to reproduce

- Ensure [build-containers](https://github.com/godotengine/build-containers) are built for 4.5-f42.
- `git clone https://github.com/graydoubt/godot-build-scripts.git`
- `cd godot-build-scripts`
- `git checkout windows-arm-only`
- copy `config.sh.in` to `config.sh`, and set `IMAGE_VERSION='4.5-f42'`
- `git clone https://github.com/godotengine/godot.git`
- `cd godot`
- switch to 4.5-beta2: `git checkout e1b4101e3`
- `cd ..`
- `./build.sh -v 4.5-beta2`



### Minimal reproduction project (MRP)

The [windows-arm-only](https://github.com/graydoubt/godot-build-scripts/tree/windows-arm-only) branch of my godot-build-scripts fork disables unwanted platforms to reduce time to failure, and mounts the `godot` directory directly, rather than expecting an archive.

--- a/thirdparty/sdl/patches/0004-errno-include.patch
+++ b/thirdparty/sdl/patches/0004-errno-include.patch
@@ -1,0 +1,13 @@
+diff --git a/thirdparty/sdl/stdlib/SDL_malloc.c b/thirdparty/sdl/stdlib/SDL_malloc.c
+index 008675f312..00118b8ae6 100644
+--- a/thirdparty/sdl/stdlib/SDL_malloc.c
++++ b/thirdparty/sdl/stdlib/SDL_malloc.c
+@@ -586,7 +586,7 @@ MAX_RELEASE_CHECK_RATE   default: 4095 unless not HAVE_MMAP
+ #define LACKS_STRING_H
+ #define LACKS_STRINGS_H
+ #define LACKS_SYS_TYPES_H
+-#define LACKS_ERRNO_H
++// #define LACKS_ERRNO_H // File uses `EINVAL` and `ENOMEM` defines, so include is required. Legacy exclusion?
+ #define LACKS_SCHED_H
+ #ifndef MALLOC_FAILURE_ACTION
+ #define MALLOC_FAILURE_ACTION

--- a/thirdparty/sdl/stdlib/SDL_malloc.c
+++ b/thirdparty/sdl/stdlib/SDL_malloc.c
@@ -586,7 +586,7 @@ MAX_RELEASE_CHECK_RATE   default: 4095 unless not HAVE_MMAP
 #define LACKS_STRING_H
 #define LACKS_STRINGS_H
 #define LACKS_SYS_TYPES_H
-#define LACKS_ERRNO_H
+// #define LACKS_ERRNO_H // File uses `EINVAL` and `ENOMEM` defines, so include is required. Legacy exclusion?
 #define LACKS_SCHED_H
 #ifndef MALLOC_FAILURE_ACTION
 #define MALLOC_FAILURE_ACTION

